shared/AppInsightsCore/src/JavaScriptSDK/CoreUtils.ts

```diff
@@ -396,6 +396,26 @@ export class CoreUtils {
         return false;
     }
 
+    /**
+     * Try to determine if the object is extensible when run within an >= ES5 container.
```

Comment:
This is the helper I added to CoreUtils based on what I think you are suggesting.